iroh-net/src/magicsock.rs

```diff
@@ -241,42 +238,777 @@ struct Inner {
     derp_map: DerpMap,
     /// Nearest DERP region ID; 0 means none/unknown.
     my_derp: AtomicU16,
+    /// Tracks the networkmap node entity for each peer discovery key.
+    peer_map: PeerMap,
+    /// UDP IPv4 socket
+    pconn4: RebindingUdpConn,
+    /// UDP IPv6 socket
+    pconn6: Option<RebindingUdpConn>,
+    /// Netcheck client
+    net_checker: netcheck::Client,
+    /// The state for an active DiscoKey.
+    disco_info: parking_lot::Mutex<HashMap<PublicKey, DiscoInfo>>,
+    udp_state: quinn_udp::UdpState,
+
+    // Send buffer used in `poll_send_udp`
+    send_buffer: parking_lot::Mutex<Vec<quinn_udp::Transmit>>,
```

Comment:
nit: I think the length of this doesn't actually change over time, so we could potentially use `Box<[Transmit]>` instead

Reply:
@Frando I filed https://github.com/quinn-rs/quinn/pull/1692 so hopefully we can get rid of this buffer entirely

Reply:
Sounds good. Let's defer to that then for now.